app/src/main/java/com/example/voyagerx/ui/fragments/landing/LandingPageFragment.kt

```diff
@@ -10,11 +11,16 @@ import com.example.voyagerx.R
 import com.example.voyagerx.databinding.FragmentLandingPageBinding
 import com.example.voyagerx.repository.LaunchRepository
 import com.example.voyagerx.repository.model.Launch
+import com.example.voyagerx.ui.fragments.landing.data.LaunchOverviewData
+import com.example.voyagerx.ui.fragments.landing.list.LaunchOverviewAdapter
+import com.example.voyagerx.ui.fragments.landing.list.LaunchOverviewClickListener
+import com.example.voyagerx.ui.fragments.landing.translators.LaunchOverviewTranslator
 
 
-class LandingPageFragment : Fragment() {
+class LandingPageFragment(private val launchOverviewTranslator: LaunchOverviewTranslator = LaunchOverviewTranslator) :
+    Fragment() {
     private lateinit var binding: FragmentLandingPageBinding
-    private lateinit var launchData: List<Launch?>
+    private lateinit var launchData: List<LaunchOverviewData>
```

Comment:
Will we still be able to send all the launch data to the details page?

Reply:
Good point. My goal with this was to map out only the data the RecyclerView needs for each card for the UI and then send the corresponding detailed info during the onClick for each item. What do you think? It might be easier to just use the whole launch object and not worry about this.

Reply:
Yeah I think it would be easier to use the whole launch object. Maybe if it feels slow we can try to refactor.